### ISSUE TITLE
[napari-worfklows] Allow label-only input

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#161]
Branching from https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/159

**Comment by tcompa** [comment]
Also note that this will be related to a different issue coming up (by @jluethi?): how could we actual pixel sizes in napari worfklows? (this concerns e.g. the use of ITK rather than regionprops).

**Comment by tcompa** [comment]
- [x] Allow label-inputs-only workflows (https://github.com/fractal-analytics-platform/fractal-tasks-core/commit/06f6725236eeedb0dd0a4ef1b9fe1bd1a5a36011).
- [ ] Add test for a workflow going from label images to measurements (without using intensity images).

**Comment by jluethi** [comment]
@tcompa Do you already have such a workflow going from label images to measurements (without using intensity images)? Otherwise, I can create one that we can use :)

**Comment by tcompa** [comment]
If you ca prepare one, even better. Thanks

**Comment by tcompa** [comment]
Re-opening since the test is still missing.

**Comment by tcompa** [comment]
Thanks @jluethi.

Writing the test brought up a different question: what should we use as a shape/chunksize of the output label?
* In the previous version (no label-only workflows allowed), we would use the property of the image (note: I am not sure the level was handled correctly).
* In the current test example (label-to-label workflow), it is clear that we should use the property of the input label also for the corresponding output label.
* How general can we make it? What if we have a multiple-labels-to-single-label workflow?

A quick solution would be to enforce this constraint:
IF there are N>=1 label outputs, THEN EITHER there are image inputs (assumed to have the same shape) OR there is one and only one label input.

Does that seem reasonable?

**Comment by jluethi** [comment]
Hmm, good scope question. I don't think label workflows without images will be very common, so I'm fine with harder limits on them. Already great we support some variants.

> IF there are N>=1 label outputs, THEN EITHER there are image inputs (assumed to have the same shape) OR there is one and only one label input.

Let's set this as the initial scope. Most of the time, one can work around this easily if necessary. As long as the error message on this is clear, I don't see this as a problem.

If someone can make a use-case that this is a common limitation, we can consider a more general solution (label chunking isn't a super critical decision. We need an answer, but if there are 2 possible answers, probably both would be fine.) But I'd see that as an optional potential update quite far down the road.

**Comment by tcompa** [comment]
Ok, I now added the constraint above ("IF there are N>=1 label outputs, THEN EITHER there are image inputs (assumed to have the same shape) OR there is one and only one label input.").

Note that I also had to add some more logic to handle the current issue:
If the was no upscaling, then also the ROI indices have to be re-built, because the pixel size that we use by default is the one from the highest-resolution images and would not work for labels where `level=0` may have a different resolution from the images' `level=0`. This is now also handled - see https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/171/commits/5dbc4e02c217be75ffb20a94369a4b2b0eac4fa9.
